src/connector/src/sink/formatter/mod.rs

```diff
@@ -106,12 +107,21 @@ impl SinkFormatterImpl {
                 )))
             }
             SinkFormat::Upsert => {
-                let key_encoder = JsonEncoder::new(
+                let mut key_encoder = JsonEncoder::new(
                     schema.clone(),
                     Some(pk_indices),
                     TimestampHandlingMode::Milli,
                 );
-                let val_encoder = JsonEncoder::new(schema, None, TimestampHandlingMode::Milli);
+                let mut val_encoder = JsonEncoder::new(schema, None, TimestampHandlingMode::Milli);
+
+                if matches!(format_desc.options.get("schema.enable"), Some(s) if s.to_lowercase() == "true")
```

Comment:
It is called `schemas.enable` (with `s`) in kafka connect and can be controlled individually on key / value. @tabVersion Could you confirm if we want this option to control both, and without `s`?